### PR TITLE
types: add Teams

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by ayntee** [comment]
Can you rename the file to `teams.ts` instead?

**Comment by itohatweb** [comment]
Renamed

File: src/types/teams.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  /** the unique id of the team */
+  id: string;
+  /** the members of the team */
+  members: TeamMemberPayload[];
+  /** the user id of the current team owner */
+  owner_user_id: string;
+}
+
+/** https://discord.com/developers/docs/topics/teams#data-models */
+export interface TeamMemberPayload {
```
Comment (ayntee) [inline src/types/teams.ts:16]:
```suggestion
export interface TeamMembersPayload {
```

File: src/types/teams.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  membership_state: MembershipState;
+  /** will always be ["*"] */
+  permissions: string[];
+  /** the id of the parent team of which they are a member */
+  team_id: string;
+  /** the avatar, discriminator, id, and username of the user */
+  user: Partial<UserPayload>;
+}
+
+/** https://discord.com/developers/docs/topics/teams#data-models */
```
Comment (ayntee) [inline src/types/teams.ts:27]:
```suggestion
/** https://discord.com/developers/docs/topics/teams#data-models-membership-state-enum */
```

File: src/types/teams.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  icon: string | null;
+  /** the unique id of the team */
+  id: string;
+  /** the members of the team */
+  members: TeamMemberPayload[];
+  /** the user id of the current team owner */
+  owner_user_id: string;
+}
+
+/** https://discord.com/developers/docs/topics/teams#data-models */
```
Comment (ayntee) [inline src/types/teams.ts:15]:
```suggestion
/** https://discord.com/developers/docs/topics/teams#data-models-team-members-object */
```

File: src/types/teams.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { UserPayload } from "./user.ts";
+
+/** https://discord.com/developers/docs/topics/teams#data-models */
```
Comment (ayntee) [inline src/types/teams.ts:3]:
```suggestion
/** https://discord.com/developers/docs/topics/teams#data-models-team-object */
```

File: src/types/teams.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+export interface TeamPayload {
+  /** a hash of the image of the team's icon */
+  icon: string | null;
+  /** the unique id of the team */
+  id: string;
+  /** the members of the team */
+  members: TeamMemberPayload[];
```
Comment (ayntee) [inline src/types/teams.ts:10]:
```suggestion
  members: TeamMembersPayload[];
```

**Review by ayntee — APPROVED** [review]
LGTM!

**Pull request merged**